jerry-core/jerryscript.h

```diff
@@ -389,6 +389,11 @@ size_t jerry_parse_and_save_literals (const jerry_char_t *source_p, size_t sourc
                                       uint32_t *buffer_p, size_t buffer_size, bool is_c_format);
 
 /**
+ * External context functions.
+ */
+size_t jerry_get_context_size (void);
+size_t jerry_get_lcache_size (size_t row_count, size_t row_length);
```

Comment:
My problem is we export internals to users, and these internals can change, while public API should not. Furthermore if I understand correctly, the problem with jerry_instance_create that it cannot allocate memory (since jerry does not depend on malloc).

Perhaps I would create a function, which has a simple configuration parameter (this is just a scratch, better names are needed):

    enum {
      type_maximum_performance,
      type_minimal_memory,
    } type;

    size_t jerry_get_instance_size(type, heap_size, index);

The heap_size is a generic parameter for the recommended heap size, but the function can change (e.g. align memory) it.

If you pass 0 as index, it return the number of spaces (e.g. 3 in the current implementation). Then you call it 3 times with index 1, 2 and 3 (type, heap_size must not be changed). You allocate memory for each size returned. And you call another function passing these pointers as an array of pointers.